prov/util/src/util_ns.c

```diff
@@ -285,15 +285,16 @@ static void *util_ns_name_server_func(void *args)
 	}
 
 	for (p = res; p; p = p->ai_next) {
-		listenfd = ofi_socket(p->ai_family, p->ai_socktype, p->ai_protocol);
+		listenfd = ofi_socket(p->ai_family, p->ai_socktype,
+				      p->ai_protocol);
 		if (listenfd != INVALID_SOCKET) {
 			n = 1;
-			setsockopt(listenfd, SOL_SOCKET, SO_REUSEADDR,
-				   &n, sizeof(n));
-			if (!bind(listenfd, p->ai_addr, p->ai_addrlen))
-				break;
-			ofi_close_socket(listenfd);
-			listenfd = INVALID_SOCKET;
+			if (setsockopt(listenfd, SOL_SOCKET,
+				       SO_REUSEADDR, &n, sizeof(n)) ||
+			    (bind(listenfd, p->ai_addr, p->ai_addrlen))) {
+				ofi_close_socket(listenfd);
+				listenfd = INVALID_SOCKET;
+			}
 		}
```

Comment:
Also the setsockopt() call is non fatal. Any error returned from that should be ignored.